aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -183,7 +183,11 @@ def _get_hashed_idempotency_key(self, lambda_event: Dict[str, Any]) -> str:
 
     @staticmethod
     def is_missing_idempotency_key(data) -> bool:
-        return data is None or not data or all(x is None for x in data)
+        if data is None or not data:
```

Comment:
if not data

Sorry Michael, I should've been clearer. Since we're not reassigning a default value, we don't need to check if data is None. If not data should account for None and falsy values

Reply:
No problem (personally i do hate Truthy stuff (but i understand some languages have it))